Adding automation groups

gtfierro opened the pull request:
Addressing #722

gtfierro committed:
adding automation groups

gtfierro committed:
remove old rec def

gtfierro committed:
fix message

gtfierro committed:
add automation group

gtfierro committed:
Merge remote-tracking branch 'origin/master' into fix-issue-722

gtfierro committed:
define rdfs

gtfierro committed:
Update examples/collection/collection.ttl

Co-authored-by: Dogan Fennibay <Dogan.Fennibay.Tr+github@ieee.org>

fennibay commented:
This is the current validation error:

```
   Severity: sh:Violation
  Focus Node: <urn:example/collection#Ahu>
  Message: Value must have at least one of the following shapes: _:7ed7414772658c5d3453cd82915382c7, _:e07496e981cc2a35110fc89f8f65c0eb
  Path: brick:hasPart
  Value: <urn:example/collection#FrostDetection>
  Source Constraint: sh:OrConstraintComponent
  Source Shape: _:ne1f5e65888c64b61a1d5168bc478d3ddb3
```

My interpretation is that we're now running against the following constraint

https://github.com/BrickSchema/Brick/blob/f7f0db97f1aecd998623c3aec2492cf3e91c16b3/bricksrc/root_class_shapes.ttl#L49-L54

But <urn:example/collection#FrostDetection> is of type brick:AutomationGroup. So I don't understand what is going wrong here.

Could it be that the SHACL engine is using a cached version or a different format of this ttl file?

gtfierro committed:
fix other hasPart constraint

gtfierro committed:
Merge branch 'fix-issue-722' of github.com:BrickSchema/Brick into fix-issue-722

gtfierro committed:
Merge remote-tracking branch 'origin/master' into fix-issue-722

gtfierro committed:
switch to poitn collection

gtfierro committed:
Merge remote-tracking branch 'origin/master' into fix-issue-722

Commit:
bump ontoenv

Commit:
Merge remote-tracking branch 'origin/master' into fix-issue-722

gtfierro committed:
some changes to automatio ncollection

gtfierro committed:
Merge branch 'fix-issue-722' of github.com:BrickSchema/Brick into fix-issue-722

gtfierro committed:
tests + changes for automation collection

gtfierro committed:
fixing sh:or constraint, rec prefix

gtfierro committed:
fix rec:includes on hvac equipment

gtfierro committed:
fix collection example

gtfierro committed:
Merge remote-tracking branch 'origin/master' into fix-issue-722

fennibay commented:
@gtfierro thank you for preparing this.

I think this is now good for the equipment part. The second aspect in #722 was collections within Zones, and I'd like address that part, as well.

A typical example is how we structure our room application. The points are organized in folders because they have some shared context, e.g.:

Z: Zone:
- C: HVAC
  - E: Radiant ceiling
- C: Room temperature setpoint determination
  - P: Room temperature setpoint
  - P: Room temperature setpoint shift
- C: Air volume flow tracking
  - P: Room supply air volume flow
  - P: Present setpoint for room extract air volume flow
- C: Ventilation control
  - P: Present ventilation setpoint
  - P: Setpoint room air quality for comfort
  - P: Setpoint room air quality for pre-comfort
  - P: Setpoint room air quality for economy

(Z: Zone, C: Collection, P: Point, E: Equipment)

Ref: <https://sid.siemens.com/r/A6V11176325/6498277643__6.0_en-US_6419929099>

So, can we apply the same principle to rec:Zone or rec:Space where an Automation_Collection can be included within them?

/cc @ektrah

Commit:
Merge remote-tracking branch 'origin/master' into fix-issue-722

Commit:
Merge branch 'fix-issue-722' of github.com:BrickSchema/Brick into fix-issue-722

gtfierro committed:
add room application example, with the zone participation in application collections

gtfierro commented:
I just implemented it, and added a quick example of your Room application: 

https://github.com/BrickSchema/Brick/blob/79fab96de19cdf0d0d442a85ca22f754d9b2ac4c/examples/collection/room_application.ttl

Let me know what you think!

fennibay commented on examples/collection/room_application.ttl:
```diff
@@ -0,0 +1,82 @@
+  rec:includes bldg:Hvac .
+
+bldg:Room a rec:Space ;
+  rec:isPartOf bldg:Zone .
```
This is not model-relevant at all, but for me: Room would be of type rec:Room and would include the zone and not vice-versa.

```suggestion
bldg:Room a rec:Room;
  rec:hasPart bldg:Zone .
```

fennibay commented on bricksrc/equipment.py:
```diff
@@ -99,7 +99,15 @@
+            ],
+            REC.includes: [
+                BRICK.HVAC_Equipment,
+                BRICK.Valve,
```
Do we need to allow includes between an equipment and its subparts? We already have hasPart for this.

For me it would be enough to just allow includes for Automation_Collection.

fennibay reviewed (changes requested):
Thank you @gtfierro, looks great for me. I have one important and one non-important comment.

gtfierro committed:
Update examples/collection/room_application.ttl

Co-authored-by: Dogan Fennibay <Dogan.Fennibay.Tr+github@ieee.org>

gtfierro committed:
do not replicate rec:includes for types we use hasPart for

gtfierro committed:
fix automatio ncollection definition

fennibay commented on bricksrc/root_class_shapes.ttl:
```diff
@@ -86,6 +91,14 @@ brick:Equipment a sh:NodeShape ;
     ] ;
     .
 
+brick:Automation_Collection a sh:NodeShape ;
+    sh:property [
+        sh:path brick:hasPart ;
+        sh:maxCount 0 ;
+        sh:message "Automation Collections must use rec:includes instead of brick:hasPart."
+    ] ;
```
I think this is colliding with the following rule: https://github.com/BrickSchema/Brick/blob/b61d72ab0b8af5ad6ff3de1320e16b2248375cb7/bricksrc/rules.ttl#L576-L598

That rule inserts a brick:hasPart relationship when there is a rec:includes relationship of a Point_Collection, and here we say that Automation_Collections may not have brick:hasPart relationships.

I would revisit rules.ttl. I think we should keep includes and hasPart separate.

fennibay commented on bricksrc/equipment.py:
```diff
@@ -99,7 +99,13 @@
+                BRICK.Valve,
+            ],
+            REC.includes: [
+                BRICK.Automation_Collection,
```
And this is colliding with <https://github.com/BrickSchema/Brick/blob/b61d72ab0b8af5ad6ff3de1320e16b2248375cb7/bricksrc/rules.ttl#L500-L526>

because that inserts rec:includes where there is a brick:hasPart and here we constrain rec:includes to brick:Automation_Collection only.

I would revisit rules.ttl

gtfierro replied:
I believe I fixed this! Thanks for the analysis

fennibay reviewed (commented):
Thanks for updating this. 

I put some analysis on the last validation errors.

gtfierro committed:
Merge remote-tracking branch 'origin/master' into fix-issue-722

gtfierro committed:
handling includes/hasPart with our new collection classes

gtfierro commented:
@fennibay I amended the documentation for this feature at https://docs.brickschema.org/modeling/point-collections.html

Please take a look and let me know  if you have any feedback!

fennibay commented:
> @fennibay I amended the documentation for this feature at https://docs.brickschema.org/modeling/point-collections.html
> 
> Please take a look and let me know if you have any feedback!

Thank you @gtfierro, looks great to me. Hopefully we can discuss this today and merge it.
